src/node.rs

```diff
@@ -155,30 +155,32 @@ impl Node {
 
         Ok(())
     }
+}
 
-    pub fn initialize_logger(&self) {
-        match self.verbosity {
-            0 => std::env::set_var("RUST_LOG", "info"),
-            1 => std::env::set_var("RUST_LOG", "debug"),
-            2 | 3 => std::env::set_var("RUST_LOG", "trace"),
-            _ => std::env::set_var("RUST_LOG", "info"),
-        };
+pub fn initialize_logger(verbosity: u8, log_sender: Option<mpsc::Sender<Vec<u8>>>) {
+    match verbosity {
+        0 => std::env::set_var("RUST_LOG", "info"),
+        1 => std::env::set_var("RUST_LOG", "debug"),
+        2 | 3 => std::env::set_var("RUST_LOG", "trace"),
+        _ => std::env::set_var("RUST_LOG", "info"),
+    };
 
-        // Filter out undesirable logs.
-        let filter = EnvFilter::from_default_env()
-            .add_directive("mio=off".parse().unwrap())
-            .add_directive("tokio_util=off".parse().unwrap())
-            .add_directive("hyper::proto::h1::conn=off".parse().unwrap())
-            .add_directive("hyper::proto::h1::decode=off".parse().unwrap())
-            .add_directive("hyper::proto::h1::io=off".parse().unwrap())
-            .add_directive("hyper::proto::h1::role=off".parse().unwrap());
-
-        // Initialize tracing.
-        let _ = tracing_subscriber::fmt()
-            .with_env_filter(filter)
-            .with_target(self.verbosity == 3)
-            .try_init();
-    }
+    // Filter out undesirable logs.
+    let filter = EnvFilter::from_default_env()
+        .add_directive("mio=off".parse().unwrap())
+        .add_directive("tokio_util=off".parse().unwrap())
+        .add_directive("hyper::proto::h1::conn=off".parse().unwrap())
+        .add_directive("hyper::proto::h1::decode=off".parse().unwrap())
+        .add_directive("hyper::proto::h1::io=off".parse().unwrap())
+        .add_directive("hyper::proto::h1::role=off".parse().unwrap());
+
+    // Initialize tracing.
+    let _ = tracing_subscriber::fmt()
+        .with_env_filter(filter)
+        .with_ansi(log_sender.is_none())
```

Comment:
`tui` can surely provide some nice colors for the logs, but this will probably need to be implemented separately, as the `ansi` feature of `tracing` doesn't translate well to `tui`'s `Paragraph`